pkg/machine/qemu/command.go

```diff
@@ -0,0 +1,105 @@
+package qemu
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/containers/podman/v4/pkg/machine"
+)
+
+// QemuCmd is an alias around a string slice to prevent the need to migrate the
+// MachineVM struct due to changes
+type QemuCmd []string
+
+// NewQemuBuilder creates a new QemuCmd object that we will build on top of,
+// starting with the qemu binary, architecture specific options, and propogated
+// proxy and SSL settings
+func NewQemuBuilder(binary string, options []string) *QemuCmd {
+	q := QemuCmd{binary}
+	q = append(q, options...)
+	return &q
+}
+
+// WithMemory adds the specified amount of memory for the machine
+func (q *QemuCmd) WithMemory(m uint64) *QemuCmd {
+	*q = append(*q, "-m", strconv.Itoa(int(m)))
```

Comment:
if you accept unit you should not cast back to int, this will just cause troubles when you overflow the type.
use `FormatUint(m, 10)`

I know pre-existing but still.